### DISCUSSION TITLE
Deprecate the PowerShell Module

### DISCUSSION THREAD

**Discussion by JanDeDobbeleer** [JanDeDobbeleer/oh-my-posh#2107] [Polls]
This discussion is aimed at getting feedback around the usage and existence of the PowerShell module as a means to install oh-my-posh. Traditionally, the module was the only way to install oh-my-posh using `Install-Module oh-my-posh`. Today, with the shift to the executable version over a year ago, it only acts as a wrapper around the executable, offering no additional functionality. Throughout the year, the following changes have been made:

- don't ship all binaries in the Module but download on `Import-Module`
- move all functionality from the Module to the [init](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/src/shell/scripts/omp.ps1) script

There's a problem with the Module due to the following:

- downloading the binary is a problem on company managed computers
- the module syncs cross device thanks to OneDrive sync, causing versions to be out of sync and [configs to break](https://ohmyposh.dev/blog/idiots-everywhere)
- it's complicated having to explain the difference time and time again (for me)

As it offers nothing extra, I'm looking for feedback if I were to remove this. 

## User impact

The migration would be rather straightforward, if you're already using the `init` style, it's only running the following and installing via [winget/scoop](https://ohmyposh.dev/docs/windows):

```powershell
Uninstall-Module oh-my-posh --AllVersions
```

If you're still using `Set-PoshPrompt -Theme ~/ myconfig.oh-my-posh.json`, same as above, but replace this line with the following:

```powershell
oh-my-posh init pwsh --config  ~/ myconfig.oh-my-posh.json | Invoke-Expression
```

**Comment by SamErde** [comment]
Have to admit I've only used the module, but hadn't referred to documentation in over a year. I'll make the switch for you!

**Reply by SamErde** [reply]
Forgot to comment back -- done! ✅

**Comment by hachi1030-Allen** [comment]
I tested the performance between PowerShell module and the scoop one, scoop one is much faster, although I don't know why.
The loading time for PowerShell module is significant compared to use oh-my-posh as a command line tool.

Therefore, go for it and it will mostly benefit the users who have been using PowerShell module.
